src/test/java/au/org/emii/ncdfgenerator/GenerationIT.java

```diff
@@ -13,69 +14,168 @@
-
-    @Before
-    public void setup() {
 
-        // TODO factor this name into var
-        new File(TMPDIR).mkdirs();
+    public int getCount() {
+        return count;
     }
+};
 
-    private NcdfEncoder getEncoder( InputStream config, String filterExpr, Connection conn ) throws Exception {
-
-        // we can't use the builder for this, becuase config is a stream...
-
-        IExprParser parser = new ExprParser();
-        IDialectTranslate translate = new PGDialectTranslate();
-        ICreateWritable createWritable = new CreateWritable(TMPDIR); // TODO factor
-        IAttributeValueParser attributeValueParser = new AttributeValueParser();
-
-        Document document = DocumentBuilderFactory.newInstance().newDocumentBuilder().parse(config);
-        Node node = document.getFirstChild();
-        NcdfDefinition definition = new NcdfDefinitionXMLParser().parse(node);
 
-        // better responsibility separation
-        // change name from outputGenerator to outputFormatter....
-        IOutputFormatter outputGenerator = new MockFormatter();
+public class GenerationIT {
 
-        return new NcdfEncoder(parser, translate, conn, createWritable, attributeValueParser, definition, filterExpr, outputGenerator, System.out);
+    static final String TMPDIR = "./tmp";
+    private static final Logger logger = LoggerFactory.getLogger(GenerationIT.class);
+
+    private static boolean dbIsPrepared = false;
+
+    public GenerationIT() throws Exception
+    {
+        if(!dbIsPrepared) {
+            logger.info("Prepare db");
+
+            InputStream is = null;
+            Connection conn = null;
+            try {
+                conn = getConn();
+
+                is = getClass().getResourceAsStream("/scripts/anmn_ts.sql");
+                conn.prepareStatement( "drop schema if exists anmn_ts cascade").execute();
+                importSQL(conn, is);
+
+                is = getClass().getResourceAsStream("/scripts/anmn_nrs_ctd_profiles.sql");
+                conn.prepareStatement( "drop schema if exists anmn_nrs_ctd_profiles cascade").execute();
+                importSQL(conn, is);
+
+                is = getClass().getResourceAsStream("/scripts/soop_sst.sql");
+                conn.prepareStatement( "drop schema if exists soop_sst cascade").execute();
+                importSQL(conn, is);
+            } finally {
+                if( is != null)
+                    is.close();
+                if( conn != null)
+                    conn.close();
+            }
+            dbIsPrepared = true;
+        }
+    }
 
+    private static void importSQL(Connection conn, InputStream is) throws Exception {
+
+        // Supports Postgres data copy actions over jdbc
+        final String regularStmts [] = {"SET", "CREATE", "COMMENT", "ALTER", "SELECT"};
+        Statement st = conn.createStatement();
+        CopyManager copyManager=((PGConnection)conn).getCopyAPI();
+
+        logger.info("Importing SQL");
+
+        while(true) {
```

Comment:
@anguss00 There was a branch/break a few lines later.  I've changed it and made the loop condition explicit since I agree it's a bit nasty.